firestore/integration_test_internal/src/aggregate_query_snapshot_test.cc

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firebase/firestore.h"
+#include "firestore_integration_test.h"
+
+#include "gtest/gtest.h"
+
+#if defined(__ANDROID__)
+#include "firestore/src/android/converter_android.h"
+#else
+#include "firestore/src/main/aggregate_query_snapshot_main.h"
+#include "firestore/src/main/converter_main.h"
+#endif  // defined(__ANDROID__)
+
+namespace firebase {
+namespace firestore {
+
+class AggregateQuerySnapshotTest : public FirestoreIntegrationTest {
+ protected:
+  static AggregateQuerySnapshot TestAggregateQuerySnapshot(
+      AggregateQuery aggregate_query, const int count) {
+    api::AggregateQuery aggregateQuery =
+        GetInternal(&aggregate_query)->aggregate_query_;
+    return MakePublic(
+        AggregateQuerySnapshotInternal(std::move(aggregateQuery), count));
+  }
+};
+
+std::size_t AggregateQuerySnapshotHash(const AggregateQuerySnapshot& snapshot) {
+  return snapshot.Hash();
+}
+
+namespace {
+
+TEST_F(AggregateQuerySnapshotTest, DefaultConstructor) {
+  AggregateQuerySnapshot snapshot;
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_FALSE(snapshot.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest, CopyConstructor) {
+  const Query& query = TestFirestore()->Collection("foo").Limit(10);
+  const AggregateQuery& aggregate_query = query.Count();
+
+  static const int COUNT = 5;
```

Comment:
`static const` local variables, although legal syntax, is highly non-idiomatic. Please just hardcode/inline the count value and maybe choose a more "unique" value, like 42 or 1234. In test code a little bit of code duplication to improve readability is fully acceptable.